cognite/client/_api/datapoints.py

```diff
@@ -96,7 +96,9 @@ def retrieve(
             limit=limit,
         )
         dps_list = fetcher.fetch(query)
-
+        for dp in dps_list:
+            dp._granularity = granularity
+            dp._cognite_client = self._cognite_client
```

Comment:
This is already being set in the DatapointsFetcher

Reply:
- it's set on the DatapointsList, but not on the Datapoints objects, and since retrieve can return one of those as well for single ids, the user can easily call to_pandas on one.